Controller/OneClick/AbandonedQuote.php

```diff
@@ -0,0 +1,370 @@
+<?php
+
+namespace Razorpay\Magento\Controller\OneClick;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Pricing\Helper\Data;
+use Razorpay\Magento\Controller\OneClick\StateMap;
+use Razorpay\Magento\Model\PaymentMethod;
+use Razorpay\Magento\Model\Config;
+use Razorpay\Api\Api;
+use Magento\Framework\App\Request\Http;
+use Magento\Directory\Model\ResourceModel\Region\CollectionFactory;
+use Magento\Directory\Model\ResourceModel\Region\Collection;
+use Razorpay\Magento\Model\CartConverter;
+use Magento\Quote\Api\CartManagementInterface;
+
+class AbandonedQuote extends Action
+{
+    /**
+     * @var Http
+     */
+    protected $request;
+
+    /**
+     * @var JsonFactory
+     */
+    protected $resultJsonFactory;
+
+    protected $collectionFactory;
+
+    /**
+     * @var Data
+     */
+    protected $priceHelper;
+
+    protected $config;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected $logger;
+
+    protected $rzp;
+
+    protected $cartRepositoryInterface;
+
+    protected $checkoutSession;
+    protected $stateNameMap;
+    protected $cartConverter;
+    protected $cartManagement;
+
+    protected $order;
+    const COD = 'cashondelivery';
+    const RAZORPAY = 'razorpay';
+    const STATE_PENDING_PAYMENT = 'pending_payment';
```

Comment:
1. `const` should be at the top of the class
2. And have their visibility `public` / `private`